app/validators/application_fulfilment_validator.rb

```diff
@@ -19,6 +19,8 @@ def validations
       ->(record) { [:children, :blank, edit_steps_children_names_path(id: '')] unless record.children.any? },
       ->(record) { [:applicants, :blank, edit_steps_applicant_names_path(id: '')] unless record.applicants.any? },
       ->(record) { [:respondents, :blank, edit_steps_respondent_names_path(id: '')] unless record.respondents.any? },
+      ->(record) { [:payment_type, :blank, edit_steps_application_payment_path] unless record.payment_type.present? },
```

Comment:
Given the submission step comes before the payment step (after we did the swap in the previous PR), I think it makes more sense to also change the order here.